toolchain/lex/lex.cpp

```diff
@@ -745,6 +793,59 @@ auto Lexer::MakeLines(llvm::StringRef source_text) -> void {
   line_index_ = 0;
 }
 
+auto Lexer::NoteWhitespace() -> void { has_leading_space_ = true; }
+
+auto Lexer::AddLexedToken(TokenInfo info) -> TokenIndex {
+  auto token = buffer_.AddToken(info);
+  has_leading_space_ = false;
+  return token;
```

Comment:
Had you considered reordering:

```suggestion
  has_leading_space_ = false;
  return buffer_.AddToken(info);
```

Reply:
Yes, went back and forth on this. I guess let's go with your suggested ordering.